settings/auth.go

```diff
@@ -0,0 +1,54 @@
+package settings
+
+import (
+	"os"
+)
+
+const (
+	EnvAuthRequired         = "AUTH_REQUIRED"
+	EnvKeycloakHost         = "KEYCLOAK_HOST"
+	EnvKeycloakRealm        = "KEYCLOAK_REALM"
+	EnvKeycloakClientID     = "KEYCLOAK_CLIENT_ID"
+	EnvKeycloakClientSecret = "KEYCLOAK_CLIENT_SECRET"
+	EnvAddonAccessSecret    = "ADDON_ACCESS_SECRET"
+)
+
+type Auth struct {
+	// Auth required
+	Required bool
+	// Keycloak client config
+	Keycloak struct {
+		Host         string
+		Realm        string
+		ClientID     string
+		ClientSecret string
+	}
+	// Addon API access secret
+	AddonAccessSecret string
```

Comment:
I assume you're using the name _AccessSecret_ to differentiate from k8s secrets.  Perhaps instead of _AccessSecret_ we could consider `AddonToken`.

Reply:
Heh, I pushed that exact change immediately before seeing your review.